### PR TITLE
Closes #1303 save pdarray parquet

### LINKED ISSUE #1303: write dataframe or multiple aligned pdarrays into parquet file(s) as a Table
I need to save multiple columns/arrays in parquet so that it can be read back as a Table. I attempted to save the pdarrays consecutively using foo.save_parquet(... mode='append'), but I kept getting errors. Even if those errors were fixed, I'm not sure it would solve my problem. I need to read the data into another application that is expecting Table objects. However, even if we ignore my specific application, I believe that most producers/consumers of parquet format files would expect Table objects rather than individual arrays.

### PR DESCRIPTION
**Pull request by Ethan-DeBandi99**
This PR (closes #1303):

`ak.DataFrame` has been updated and now includes a `save_table()` method. This method works by calling `save_all()` on the data mapping. The `Index` is added to the data mapping if a users sets `index=True`.

`pdarrayIO.save_all()` has been updated to now allow saving as parquet or hdf5. This allows the user to save a list of `pdarray` to a file. Testing only provided for Parquet at this time. `pd.read_parquet()` allows use to provide a directory to read all the files in for multilocale. 

Everything is configured so that once strings is updated to use the new `file_format` parameter, that will be functional for dataframe saving as well.

### CODE REVIEW

**Comment by Ethan-DeBandi99** [comment]
Tests that are currently broken will be resolved with `Strings.save()` update to support new configuration. Keeping this PR in draft until PR #1307 is merged.

File: arkouda/dataframe.py [hunk 1]
```diff
@@ -12,7 +12,8 @@
 from arkouda.pdarraycreation import arange, array
 from arkouda.groupbyclass import GroupBy as akGroupBy
 from arkouda.pdarraysetops import concatenate, unique, intersect1d, in1d
-from arkouda.pdarrayIO import save_all
+from arkouda.pdarrayIO import save_all, load_all
+from arkouda.pdarrayIO import save_all, load_all
```
Comment (joshmarshall1) [inline arkouda/dataframe.py:16]: Duplicate import

**Review by joshmarshall1 — APPROVED** [review]
All looks good except a duplicated `from arkouda.pdarrayIO import save_all, load_all` in `dataframe.py`

File: arkouda/dataframe.py [hunk 2]
```diff
@@ -1068,6 +1069,41 @@ def save(self, path, index=False):
...
+        columns: List
+            List of columns to include in the file. If None, writes out all columns
+        file_format: str
+            'HDF5' or 'Parquet'. Defaults to 'HDF5'
+        index: Bool
+            If true, include the index values in the save file.
+
+        Notes
+        ______
+        This function current uses 'truncate' mode to ensure the file exists before appending.
```
Comment (pierce314159) [inline arkouda/dataframe.py:1089]: `current` > `currently`

File: arkouda/dataframe.py [hunk 3]
```diff
@@ -1068,6 +1069,41 @@ def save(self, path, index=False):
...
+        if index:
+            data["Index"] = self.index
+        save_all(data, prefix_path=prefix_path,
+                 file_format=file_format)
+    @classmethod
+    def load_table(cls, prefix_path):
+        return cls(load_all(prefix_path))
+
+
+
```
Comment (pierce314159) [inline arkouda/dataframe.py:1106]: this is just white space so but we have 0 newlines between `save_table` & `load_table` but 3 between `load_table` & `argsort`

File: tests/io_test.py [hunk 4]
```diff
@@ -529,6 +529,7 @@ def testAppendMixedStringsDataset(self):
         
         strings = strings_array.to_ndarray()
         strings.sort()
+        print(r_mixed)
```
Comment (pierce314159) [inline tests/io_test.py:532]: stray print statement

**Review by pierce314159 — APPROVED** [review]
Everything looks good! Couple small formatting things

File: tests/io_test.py (hunk 4 above)
Comment (Ethan-DeBandi99) [reply tests/io_test.py]: Removed

File: arkouda/dataframe.py (hunk 3 above)
Comment (Ethan-DeBandi99) [reply arkouda/dataframe.py]: Updated. Good catch.

File: arkouda/dataframe.py (hunk 1 above)
Comment (Ethan-DeBandi99) [reply arkouda/dataframe.py]: Removed duplicate import

File: arkouda/dataframe.py (hunk 2 above)
Comment (Ethan-DeBandi99) [reply arkouda/dataframe.py]: Updated.

**Pull request merged**